# Statistika Deskriptif - Mean dan Median

Identitas diri : Ruth Septriana Sipangkar_24060124120024 (E)

Untuk mempelajari lebih lanjut mengenai properti dari mean dan median, kita gunakan library Panda dan dataset berikut.

In [ ]:
import pandas as pd

# Unduh file CSV jika belum ada di direktori saat ini
!pip install gdown
!gdown --id 1KEwyokOzuHi28Td8vZETxstNlmMRE8dC

# Membaca file CSV yang ada di direktori sebagai DataFrame
df = pd.read_csv("Latihan.csv")

Dataset "Latihan" merupakan dataset yang berisi sampel jumlah jawaban benar untuk masing-masing tipe ujian dan skornya pada suatu latihan suatu mata kuliah tertentu.

## Properti Asli

Mari kita hitung nilai mean dan median dari data frame "Latihan", untuk kali ini saya akan menggunakan dataset **Tipe 1**.

In [11]:
# Menampilkan 5 baris pertama dari DataFrame
print(df.head())

# Menghitung mean dan median untuk dataset "Tipe 1"
mean_tipe1 = df['Tipe1'].mean()
median_tipe1 = df['Tipe1'].median()

# Menampilkan hasil mean dan median
print("Mean dari dataset Tipe 1:", mean_tipe1)
print("Median dari dataset Tipe 1:", median_tipe1)

   Tipe1  Tipe2  Tipe3  Tipe4
0      5      5      4      2
1      5      3      2      3
2      5      4      2      3
3      5      3      2      2
4      4      3      2      4
Mean dari dataset Tipe 1: 7.839285714285714
Median dari dataset Tipe 1: 7.0


## Properti 1: Menambah atau Mengurangi semua Dataset dengan Nilai Konstanta

### Langkah 1: Memodifikasi dataset

Misal kita lakukan modifikasi kepada setiap data di dataset dengan **penjumlahan 2** sebagai modifikasi1 dan **pengurangan 3** sebagai modifikasi2.

In [12]:
# Menambahkan 2 ke setiap elemen sebagai modifikasi1
df_modif1 = df+2
# Menampilkan 5 baris pertama dari dataset modifikasi1
print("Dataset modifikasi1:")
print(df_modif1.head())

# Mengurangkan 3 ke setiap elemen sebagai modifikasi2
df_modif2 = df-3
# Menampilkan 5 baris pertama dari dataset modifikasi2
print("\nDataset modifikasi2:")
print(df_modif2.head())

Dataset modifikasi1:
   Tipe1  Tipe2  Tipe3  Tipe4
0      7      7      6      4
1      7      5      4      5
2      7      6      4      5
3      7      5      4      4
4      6      5      4      6

Dataset modifikasi2:
   Tipe1  Tipe2  Tipe3  Tipe4
0      2      2      1     -1
1      2      0     -1      0
2      2      1     -1      0
3      2      0     -1     -1
4      1      0     -1      1


### Langkah 2: Menghitung nilai mean dan median dari data yang sudah dimodifikasi

In [13]:
# Menghitung mean dan median untuk modifikasi1 di dataset "Tipe 1"
mean_modif1 = df_modif1['Tipe1'].mean()
median_modif1 = df_modif1['Tipe1'].median()
# Menampilkan hasil mean dan median
print("Mean dari modifikasi 1 dataset Tipe 1:", mean_modif1)
print("Median dari modifikasi 1 dataset Tipe 1:", median_modif1)

# Menghitung mean dan median untuk modifikasi2 di dataset "Tipe 1"
mean_modif2 = df_modif2['Tipe1'].mean()
median_modif2 = df_modif2['Tipe1'].median()
# Menampilkan hasil mean dan median
print("\nMean dari modifikasi 2 dataset Tipe 1:", mean_modif2)
print("Median dari modifikasi 2 dataset Tipe 1:", median_modif2)

Mean dari modifikasi 1 dataset Tipe 1: 9.839285714285714
Median dari modifikasi 1 dataset Tipe 1: 9.0

Mean dari modifikasi 2 dataset Tipe 1: 4.839285714285714
Median dari modifikasi 2 dataset Tipe 1: 4.0


### Langkah 3: Analisis

Dari beberapa perhitungan diatas kita peroleh hasil sebagai berikut:

Properti asli                : mean 7.83 dan median 7

Modifikasi 1 (penjumlahan 2) : mean 9.83 dan median 9 --> ($9.83 = 7.83 + 2$ dan $9 = 7 + 2$)

Modifikasi 2 (pengurangan 3) : mean 4.83 dan median 4 --> ($4.83 = 7.83 - 3$ dan $4 = 7 - 3$)

Jika melihat polanya, bisa kita simpulkan bahwa mean dan median dari penjumlahan (atau pengurangan) konstanta kepada setiap data dalam dataset sama dengan mean dan median dari dataset ditambah (atau dikurang) konstanta tersebut.

Atau bisa dituliskan secara matematis sebagai berikut:

$$
\frac{\sum x_i \pm c}{n} = \frac{\sum x_i}{n} \pm c
$$

$$
\bar{x}_{\text{baru}} = \bar{x} \pm c
$$

Pada median, sebagai berikut:

$$
\text{Me}_{\text{baru}} = \text{Me} \pm c
$$




## Properti 2: Mengalikan atau Membagi semua Dataset dengan Nilai Konstanta

### Langkah 1: Memodifikasi dataset

Misal kita lakukan modifikasi kepada setiap data di dataset dengan **perkalian 2** sebagai modifikasi1 dan **pembagian 3** sebagai modifikasi2.

In [14]:
# Menglikan 2 ke setiap elemen sebagai modifikasi1
df_modif1 = df*2
# Menampilkan 5 baris pertama dari dataset modifikasi1
print("Dataset modifikasi1:")
print(df_modif1.head())

# Membagi 3 ke setiap elemen sebagai modifikasi2
df_modif2 = df/3
# Menampilkan 5 baris pertama dari dataset modifikasi2
print("\nDataset modifikasi2:")
print(df_modif2.head())

Dataset modifikasi1:
   Tipe1  Tipe2  Tipe3  Tipe4
0     10     10      8      4
1     10      6      4      6
2     10      8      4      6
3     10      6      4      4
4      8      6      4      8

Dataset modifikasi2:
      Tipe1     Tipe2     Tipe3     Tipe4
0  1.666667  1.666667  1.333333  0.666667
1  1.666667  1.000000  0.666667  1.000000
2  1.666667  1.333333  0.666667  1.000000
3  1.666667  1.000000  0.666667  0.666667
4  1.333333  1.000000  0.666667  1.333333


### Langkah 2: Menghitung nilai mean dan median dari data yang sudah dimodifikasi

In [15]:
# Menghitung mean dan median untuk modifikasi1 di dataset "Tipe 1"
mean_modif1 = df_modif1['Tipe1'].mean()
median_modif1 = df_modif1['Tipe1'].median()
# Menampilkan hasil mean dan median
print("Mean dari modifikasi 1 dataset Tipe 1:", mean_modif1)
print("Median dari modifikasi 1 dataset Tipe 1:", median_modif1)

# Menghitung mean dan median untuk modifikasi2 di dataset "Tipe 1"
mean_modif2 = df_modif2['Tipe1'].mean()
median_modif2 = df_modif2['Tipe1'].median()
# Menampilkan hasil mean dan median
print("\nMean dari modifikasi 2 dataset Tipe 1:", mean_modif2)
print("Median dari modifikasi 2 dataset Tipe 1:", median_modif2)



Mean dari modifikasi 1 dataset Tipe 1: 15.678571428571429
Median dari modifikasi 1 dataset Tipe 1: 14.0

Mean dari modifikasi 2 dataset Tipe 1: 2.613095238095238
Median dari modifikasi 2 dataset Tipe 1: 2.3333333333333335


### Langkah 3: Analisis

Dari beberapa perhitungan diatas kita peroleh hasil sebagai berikut:

Properti asli                : mean 7.83 dan median 7

Modifikasi 1 (perkalian 2) : mean 15.67 dan median 14 --> ($15.67 = 7.83\times2$ dan $14 = 7\times2$)

Modifikasi 2 (pembagian 3) : mean 2.61 dan median 2.33 --> ($2.61 = 7.83/3$ dan $2.33 = 7/3$)

Jika melihat polanya, bisa kita simpulkan bahwa mean dan median dari perkalian (atau pembagian) konstanta kepada setiap data dalam dataset sama dengan mean dan median dari dataset dikali (atau dibagi) konstanta tersebut.

Atau bisa dituliskan secara matematis sebagai berikut:

$$
\frac{\sum x_i \times c}{n} = \frac{\sum x_i}{n} \times c
$$

$$
\bar{x}_{\text{baru}} = \bar{x} \times c
$$

Pada median, sebagai berikut:

$$
\text{Me}_{\text{baru}} = \text{Me} \times c
$$



## Properti 3: Sensitif terhadap Outlier

### Langkah 1: Memodifikasi dataset

Misal kita lakukan modifikasi dimana terdapat mahasiswa baru super jenius yang mendapat nilai 1000 di tiap tipe ujian, kita tambahkan **1 baris data dengan nilai 1000** ke dalam dataset.

In [16]:
# Menambahkan satu baris data dengan nilai 1000 untuk setiap kolom tipe ujian
new_row = pd.DataFrame({col: [1000] for col in df.columns})
df = pd.concat([df, new_row], ignore_index=True)

# Menghapus baris terakhir yang ditambahkan
# Gunakan agar tidak ada baris baru ganda jika program dijalankan berulang kali
# df = df.iloc[:-1] 

# Menampilkan 5 baris terakhir dari dataset modifikasi outlier
print("\nDataset modifikasi outlier:")
print(df.tail())


Dataset modifikasi outlier:
    Tipe1  Tipe2  Tipe3  Tipe4
52     15     10      8      2
53      4     20     12      5
54     15      6      5      4
55     20      5      3      6
56   1000   1000   1000   1000


### Langkah 2: Menghitung nilai mean dan median dari data yang sudah dimodifikasi

In [17]:
# Menghitung mean dan median untuk modifikasiol di dataset "Tipe 1"
mean_modifol = df['Tipe1'].mean()
median_modifol = df['Tipe1'].median()
# Menampilkan hasil mean dan median
print("Mean dari modifikasi outlier dataset Tipe 1:", mean_modifol)
print("Median dari modifikasi outlier dataset Tipe 1:", median_modifol)

Mean dari modifikasi outlier dataset Tipe 1: 25.24561403508772
Median dari modifikasi outlier dataset Tipe 1: 7.0


### Langkah 3: Analisis

Dari beberapa perhitungan diatas kita peroleh hasil sebagai berikut:

Properti asli      : mean 7.83 dan median 7

Modifikasi outlier : mean 25.24 dan median 7

Dari data tersebut bisa kita lihat bahwa mean berubah sangat drastis, sedangkan median tidak berubah. Ini karena mean menjumlahkan seluruh data dan membaginya dengan jumlah data. Penambahan outlier akan menghasilkan nilai penjumlahan seluruh data yang sangat besar, sedangkan pembaginya hanya bertambah satu, sehingga hasil pembagian akan sangat besar. Sedangkan median hanya mengambil data di tengah dataset, sehingga outlier bisa dihiraukan.

Dari sini bisa kita simpulkan bahwa outlier sangat mempengaruhi mean, sedangkan untuk median tidak.


